ldi-nifi/ldi-nifi-processors/ldes-client-processor/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/processors/LdesClient.java

```diff
@@ -57,87 +60,76 @@ public Set<Relationship> getRelationships() {
 	@Override
 	public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
 		return List.of(DATA_SOURCE_URL, DATA_SOURCE_FORMAT, DATA_DESTINATION_FORMAT, FRAGMENT_EXPIRATION_INTERVAL,
-				STREAM_TIMESTAMP_PATH_PROPERTY, STREAM_VERSION_OF_PROPERTY, STREAM_SHAPE_PROPERTY);
-	}
-
-	@OnAdded
-	public void onAdded() {
-		if (identifier == null) {
-			identifier = getIdentifier();
-		}
-		config.setPersistenceDbName(identifier + "-" + config.getPersistenceDbName());
+				STREAM_TIMESTAMP_PATH_PROPERTY, STREAM_VERSION_OF_PROPERTY, STREAM_SHAPE_PROPERTY, OAUTH_CLIENT_ID,
+				OAUTH_CLIENT_SECRET, OAUTH_TOKEN_ENDPOINT, OAUTH_SCOPE);
 	}
 
 	@OnScheduled
 	public void onScheduled(final ProcessContext context) {
```

Comment:
I feel like we have to know a lot about how to wire the client before we can use it. Maybe we should think of a way that we can pass all our configuration to one initializer/factory in the client that knows how to put everything together and provide a fully configured memberSupplier that we can then use?